web/src/components/CustomMarquee/CustomMarquee.jsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import Marquee from "react-fast-marquee";
+
+import PropTypes from "prop-types";
+
+function CustomMarquee(props) {
+    const {
+        data,
+        containerRef,
+    } = props;
+
+    const [windowSize, setWindowSize] = React.useState(window.innerWidth);
+    const [resizeCooldown, setResizeCooldown] = React.useState(false);
+    function handleResize(){
+        if(!resizeCooldown && window.innerWidth != windowSize){
+            setWindowSize(window.innerWidth);
+            setResizeCooldown(true);
+
+            setTimeout(()=>{setResizeCooldown(false);}, 1000) // set a cooldown for resize checks to avoid excessive renders
+        }
+    }
+    window.addEventListener("resize", handleResize());
+
+    const dataRef = React.useRef(null);
+    const [dataScroll, setDataScroll] = React.useState(false);
+    const [dataOverrideTimeout, setDataOverrideTimeout] = React.useState(null);
+    const [dataScrollOverride, setDataScrollOverride] = React.useState(true); // set to true by defualt so that there is no scrolling on load, until a setTimeout later on changes that 2 seconds after load
+
+    React.useEffect(() => {
+        const container = containerRef.current;
+        const scroll = dataRef.current;
+
+        if(container && scroll){
+            if(container.offsetWidth < scroll.scrollWidth && !dataScrollOverride){
+                console.log("Adding Marquee to " + data);
+                setDataScroll(true);
+            } else if(container.offsetWidth >= scroll.scrollWidth && dataScrollOverride){ // dataScroll is only able to be set to false when a scroll has been completed to avoid randomly jumbled text
+                clearTimeout(dataOverrideTimeout)
+            }
+        }
+    }, [data, dataScrollOverride, windowSize]) // Assesses need to scroll whenever the song changes, the scroll completes, or when the window resizes
+
+    function setOverride(){
+        setDataScroll(false);
+        setDataScrollOverride(true);
+
+        setDataOverrideTimeout(setTimeout(()=>{setDataScroll(true); setDataScrollOverride(false);}, 2000)); // When a scroll completes, pause for two seconds and restart (unless the timeout is cleared)
+    }
+
+    setTimeout(()=>{setDataScrollOverride(false);}, 2000) // used to kick off useEffect after two seconds of page load
+
+    return(
+        <Marquee play={dataScroll} speed={30} delay={2} onCycleComplete={() => {setOverride()}}>
```

Comment:
you dedicate a lot of state to managing a scrolling delay, but your upstream Marquee library already has a `delay` param. why is your delay state necessary? can it be removed, such that we don't trigger so many re-renders with changes in `useState()` things?

Reply:
https://www.react-fast-marquee.com/documentation#delay
The built-in delay only works on initial render, not on every cycle completion